launch.go

```diff
@@ -182,11 +182,23 @@ func launch(api screwdriver.API, buildID string, rootDir string) error {
 		j.Name = "main"
 	}
 
-	err = gitSetup(scm.httpsString(), w.Src, pr, b.SHA)
+	repo, err := newRepo(scm.httpsString(), w.Src)
 	if err != nil {
 		return err
 	}
 
+	err = repo.Checkout(b.SHA)
+	if err != nil {
+		return err
+	}
+
+	if pr != "" {
+		err = repo.MergePR(pr, "pr")
```

Comment:
`"pr"` here seems unnecessary. Can we just move it into MergePR and make it a single parameter?